samtranslator/model/api/api_generator.py

```diff
@@ -576,7 +598,39 @@ def _construct_api_domain(  # noqa: too-many-branches
 
             record_set_group.RecordSets += self._construct_record_sets_for_domain(self.domain, api_domain_name, route53)
 
-        return domain, basepath_resource_list, record_set_group
+        return ApiDomainResponse(domain, basepath_resource_list, record_set_group, None)
+
+    def _construct_individual_record_set(
+        self, domain: Dict[str, Any], api_domain_name: str, route53: Any
+    ) -> List[Route53RecordSet]:
+        logical_id_suffix = LogicalIdGenerator(
+            "", [route53.get("HostedZoneId") or route53.get("HostedZoneName"), domain.get("DomainName")]
+        ).gen()
+        logical_id = "RecordSet" + logical_id_suffix
+        logical_id_ipv6 = "RecordSetIpv6" + logical_id_suffix
+
+        recordset_list = []
+
+        recordset = Route53RecordSet(logical_id, attributes=self.passthrough_resource_attributes)
+        recordset.Name = domain.get("DomainName")
+        recordset.Type = "A"
+        recordset.AliasTarget = self._construct_alias_target(domain, api_domain_name, route53)
+
+        if route53.get("HostedZoneId"):
+            recordset.HostedZoneId = route53.get("HostedZoneId")
+        else:
+            recordset.HostedZoneName = route53.get("HostedZoneName")
+
+        recordset_list.extend([recordset])
+
+        if route53.get("IpV6"):
+            recordset_ipv6 = Route53RecordSet(logical_id_ipv6, attributes=self.passthrough_resource_attributes)
```

Comment:
Thoughts on having this as part of the `RecordSetGroup` for the domain (i.e. both IPv4 and IPv6 under same group)? Seems neat, and might help with preventing hitting the 500-resource limit.

Reply:
Synced offline. Decided to use `RecordSetGroup` to combine IPv4/IPv6